engine/src/main/java/org/terasology/rendering/dag/stateChanges/SwapGBuffers.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.stateChanges;
+
+import org.terasology.rendering.dag.StateChange;
+import org.terasology.rendering.opengl.SwappableFBO;
+
+/**
+ * This StateChange is used to swap the readOnly and writeOnly buffers in the gBufferPair.
+ * This is useful for Nodes that want to update the value of the gBuffer, by passing it's contents through a shader.
+ *
+ * Note that this StateChange is special because unlike all the other StateChanges that modify the state of the
+ * system during the execution of the task list, this StateChange does it's job by indirectly changing what data gets
+ * sent to future StateChanges, by causing a change in the return value of gBufferPair.get*Buffer().
+ *
```

Comment:
I'm thinking of changing the middle paragraph to:

` * Note that this StateChange is special because unlike all the other StateChanges that modify the state of the
 * system during the execution of the task list, this StateChange does it's job during the initialization phase,
 * changing the final state of the system indirectly by changing what data gets sent to future StateChanges.
 * It does so by causing a change in the return value of gBufferPair.get*Buffer().`